source/common/upstream/logical_dns_cluster.cc

```diff
@@ -21,8 +21,7 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
   }
 
   dns_url_ = hosts_json[0]->getString("url");
-  Network::Utility::hostFromTcpUrl(dns_url_);
-  Network::Utility::portFromTcpUrl(dns_url_);
+  hostname_ = Network::Utility::hostAndPortFromTcpUrl(dns_url_);
```

Comment:
There are 2 things being conflated here, how Envoy parses "URLs" and what a web/proxy server expects in the Host/:authority header. This change is about auto-populating the Host/:authority header. Technically, all web/proxy servers should support host:port when doing vhost, etc. matching, but that doesn't always work, and as I said above, Envoy itself is broken in this regard currently. I think you will have better luck with this feature if you just populate the DNS host name and skip the port.